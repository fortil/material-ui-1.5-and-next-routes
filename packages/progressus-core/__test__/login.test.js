/* eslint-env jest */
/* 
  @By William Penagos billalpeza@gmail.com
*/
import { shallow, mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import App from '../login'
const Login = App()

describe('Functions login testing', () => {
  describe('With params', () => {
    const onChange = jest.fn()
    const login = jest.fn()
    const app = mount(<Login change={onChange} login={login} />)
    const email = 'william.penagos@sourcemeridian.com'
    const pwd = '123sourcemeridian.com'

    it('email', () => {
      app.find('TextField').find('input').at(0).props().onChange({ target: { value: email } })
      expect(onChange).toBeCalledWith({ prop: 'user', value: email })
    })
    it('password', () => {
      app.find('Input#password').find('input').props().onChange({ target: { value: pwd } })
      expect(onChange).toBeCalledWith({ prop: 'password', value: pwd })
    })
    it('submit', () => {
      app.find('TextField').find('input').at(0).props().onChange({ target: { value: email } })
      app.find('Input#password').find('input').at(0).props().onChange({ target: { value: pwd } })
      app.find('Button').find('button').at(0).simulate('click')
      expect(login).toBeCalledWith({ user: email, password: pwd })
    })
  })
  describe('Without params', () => {
    const app = shallow(<Login />).dive()
    const email = 'william.penagos@sourcemeridian.com'
    const pwd = '123sourcemeridian.com'

    it('email', () => {
      app.find('TextField').simulate('change', { target: { value: email } })
      expect(app.state('user')).toEqual(email)
    })
    it('password', () => {
      app.find('#password').simulate('change', { target: { value: pwd } })
      expect(app.state('password')).toEqual(pwd)
    })
  })
})

describe('With Snapshot Testing', () => {
  it('Snapshot of login', () => {
    const component = renderer.create(<Login />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
