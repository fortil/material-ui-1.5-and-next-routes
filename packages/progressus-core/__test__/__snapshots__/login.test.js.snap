// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Snapshot of login 1`] = `
<div
  className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiPaper-rounded-14 Login-paper-1"
>
  <div
    className="MuiGrid-container-40 MuiGrid-align-items-xs-center-48 MuiGrid-justify-xs-center-57"
  >
    <div
      className="MuiGrid-item-41 MuiGrid-grid-md-4-100 Login-image-3"
    >
      <img
        alt="PROGRESUS"
        src="http://lorempixel.com/400/200"
        style={
          Object {
            "width": "100%",
          }
        }
      />
    </div>
    <div
      className="MuiGrid-item-41 MuiGrid-grid-md-8-104"
    >
      <div
        className="MuiGrid-container-40 MuiGrid-direction-xs-column-43 MuiGrid-align-items-xs-center-48 MuiGrid-justify-xs-center-57"
      >
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 Login-titleContainer-4"
        >
          <h2
            className="MuiTypography-root-137 MuiTypography-title-143 Login-title-5"
          >
            Inicio Sesión
          </h2>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 Login-inputUserContainer-6"
        >
          <div
            className="MuiFormControl-root-162 Login-inputUser-7"
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
            >
              Usuario
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 Login-inputPasswordContainer-8"
        >
          <div
            className="MuiFormControl-root-162 Login-margin-9 Login-textField-10"
          >
            <label
              className="MuiFormLabel-root-171 MuiInputLabel-root-166 MuiInputLabel-formControl-167 MuiInputLabel-animated-170"
              data-shrink={false}
              htmlFor="password"
            >
              Contraseña
            </label>
            <div
              className="MuiInput-root-178 MuiInput-formControl-179 MuiInput-underline-182"
            >
              <input
                aria-invalid={false}
                className="MuiInput-input-186 MuiInput-inputType-189"
                disabled={false}
                id="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
                value=""
              />
              <div
                className="MuiInputAdornment-root-191 MuiInputAdornment-positionStart-192"
              >
                <button
                  aria-label="Toggle password visibility"
                  className="MuiButtonBase-root-200 MuiIconButton-root-194"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="button"
                >
                  <span
                    className="MuiIconButton-label-199"
                  >
                    <svg
                      aria-hidden="true"
                      className="MuiSvgIcon-root-203"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                  </span>
                  <span
                    className="MuiTouchRipple-root-210"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 Login-buttonContainer-11"
        >
          <button
            className="MuiButtonBase-root-200 MuiButton-root-217 MuiButton-contained-228 MuiButton-containedPrimary-229 MuiButton-raised-231 MuiButton-raisedPrimary-232 Login-button-12"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-218"
            >
              Ingresar
            </span>
            <span
              className="MuiTouchRipple-root-210"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
